### PR TITLE
Include @types/google-protobuf as a dependency

### PR DESCRIPTION
**Pull request by blampe**
# Description

Fixes https://github.com/pulumi/pulumi/issues/14781

## Checklist

- [ ] I have run `make tidy` to update any new dependencies
- [ ] I have run `make lint` to verify my code passes the lint check
  - [ ] I have formatted my code using `gofumpt`

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: .github/workflows/pr-test-codegen-downstream.yml [hunk 1]
```diff
@@ -66,6 +66,7 @@ jobs:
       needs: ["cleanup"]
       timeout-minutes: 240
       runs-on: ubuntu-latest
+      if: false # TODO(https://github.com/pulumi/action-test-provider-downstream/issues/20)
```
Comment (justinvp) [inline .github/workflows/pr-test-codegen-downstream.yml:69]: Can you delete this change?

File: cmd/pulumi-test-language/interface.go [hunk 2]
```diff
@@ -262,8 +265,10 @@ func compareDirectories(actualDir, expectedDir string, allowNewFiles bool) ([]st
 		}
 
 		if !bytes.Equal(actualContents, expectedContents) {
-			// TODO(https://github.com/pulumi/pulumi/issues/13943): Find a way to show better diffs here
-			validations = append(validations, fmt.Sprintf("expected file %s does not match actual file", relativePath))
+			edits := myers.ComputeEdits(span.URIFromPath("expected"), string(expectedContents), string(actualContents))
```
Comment (justinvp) [inline cmd/pulumi-test-language/interface.go:268]: This seems unrelated. Can you remove the changes in this file and do it as a separate change.?

File: changelog/pending/20231207--sdkgen-nodejs--include-attypes-google-protobuf-for-sdk-codegen.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+changes:
+- type: fix
+  scope: sdkgen/nodejs
+  description: Include @types/google-protobuf for SDK codegen
```
Comment (justinvp) [inline changelog/pending/20231207--sdkgen-nodejs--include-attypes-google-protobuf-for-sdk-codegen.yaml:4]:
```suggestion
  description: Include @types/google-protobuf as a dependency
```

File: pkg/codegen/nodejs/gen.go [hunk 4]
```diff
@@ -14,7 +14,7 @@
 
 // Pulling out some of the repeated strings tokens into constants would harm readability, so we just ignore the
 // goconst linter's warning.
-//
```
Comment (justinvp) [inline pkg/codegen/nodejs/gen.go:17]: Remove changes to this file

**Review by justinvp — CHANGES_REQUESTED** [review]

File: cmd/pulumi-test-language/interface.go (hunk 2 above)
Comment (blampe) [reply cmd/pulumi-test-language/interface.go]: 👍 https://github.com/pulumi/pulumi/pull/14786

File: .github/workflows/pr-test-codegen-downstream.yml (hunk 1 above)
Comment (blampe) [reply .github/workflows/pr-test-codegen-downstream.yml]: Will this work? We'll find out!
